client/packages/requisitions/src/RequestRequisition/api/operations.graphql

```diff
@@ -458,16 +458,18 @@ mutation deleteRequest(
 }
 
 fragment ProgramSettings on ProgramRequisitionSettingNode {
-  program {
-    name
-    suppliers {
-      name
-    }
+  label: programName
```

Comment:
not a fan of renaming data to match the expected format of the consumer component.
I would change the Autocomplete/types.ts to 
```
export type AutocompleteOption<SomeObject> = SomeObject & { label?: string };
```
have these fields as `name` 
and add this to the Autocomplete used in LabelAndOptions

```
getOptionLabel={option => {
              const o = option as { programName?: string; name?: string };
              return ('programName' in o ? o.programName : o.name) ?? '';
            }}
```

and make label optional in `LabelAndOptions` to avoid the ts complaint

though this would be easier if the component wasn't reused for all of the types, or if `programName` was just `name` 🤷

Reply:
> not a fan of renaming data to match the expected format of the consumer component.

Hmm, but we do this quite a lot for AutocompleteOption, do you mean it's the distance that matters in this case ? (we usually do it close to AutocompleteOption), and here it's done in grahpQL ? I am actually not a fan how we do it usually, and I think we should always to it with getOptionLabel or even better `optionKey` parameter (this way onChange preserves the original data type).

> if programName was just name

I think programName should still remain (in API), since `name` on ProgramRequisitionSettings is ambiguous. How does this look ?

> export type AutocompleteOption<SomeObject> = SomeObject & { label?: string };

I did it [here (downstream, with optionKey)](https://github.com/openmsupply/open-msupply/commit/41a035565b9f0a4b3926e949ffe9e47e39edfefe) but I don't think it's safe in general, it would mean if label field is missing, and mapping is not defined you would need to run the app to validate if it works (I think we should maximise typescript and type safety to reduce amount of manual tests we need to do, i.e. compulsory union of either getOptionLabel, optionKey or renderOption)